packages/backend/src/queryBuilder.ts

```diff
@@ -307,10 +386,14 @@ export const applyLimitToSqlQuery = ({
     limit: number | undefined;
 }): string => {
     if (limit === undefined) return sqlQuery;
-    return `WITH user_sql AS (\n${sqlQuery.replace(
-        /;\s*$/,
-        '',
-    )}\n) select * from user_sql limit ${limit}`;
+    const existingLimit = extractOuterLimitFromSQL(sqlQuery);
+    if (existingLimit !== undefined && existingLimit <= limit) {
+        // existing limit is less than or equal to the provided limit
+        return sqlQuery;
+    }
+    // remove existing limits and apply the new limit
+    const sqlWithoutCommentsAndLimits = removeCommentsAndLimits(sqlQuery);
+    return `WITH user_sql AS (\n${sqlWithoutCommentsAndLimits}\n) select * from user_sql limit ${limit}`;
```

Comment:
The initial order problem is caused because of this logic that wraps their SQL in a CTE. 
So we need to add a `LIMIT ${limit}` at the end of their query instead.